README.md
# Build

To build all addons:

```
.\build.ps1
```

To build a single specific addon:

```
.\build.ps1 -only helmets
```

To build multiple specific addons:

```
.\build.ps1 -only helmets, weapons
```


changelog.txt
CHANGELOG 5/13/21 - The Rebalance Update

Major Restructuring of Mod into separate parts, to accelerate further development, testing, and changes.

101st Aux LAATs
	Change of lockon sounds
	Change of Cockpit/Interior sounds to be quieter

101st Holosights
	change of sights position for the following weapons (better sight visibility, and matches where sights should actually be)
		-Zakkeg
		-DC15S
		-DC15S(Sidearm)
		-DC17
		-SM-01 PDW
		-PDW Defender

Helmets
	Updated Shotty's Helmet
	Fixes for the Tech Helmet

101st Weapons
	Removed Dependancy on Legacy RD501 Ammunition for small arms

Preparation Files for Future MBT and Manadalorian Ship projects.



CHANGELOG 12.7.20
	
101st Aux LAATs
	Slightly reduced systems armor
	Greatly reduced Air to ground munitions to encourage use of VTOLs
	Adding Searchlight LAAT
	Adding Command versions of LAAT
	Added RD501 Laat Switcher, LAAT comes in many flavors now.
	Reduced Mk3's Armor to be equivalent to Mk5/6

101st Nu Class Shuttle
	Fixed texture to be correct green, still has issues but is better now.
	Increased armor to 1300 as testbed, systems are fragile, however hull is not. Will be able to shrug off several hits before being disabled, and even when disabled, will be able to land safely. (Tested 1 King AAT shot disables, 6 shots to destroy)
	Added Impulse

15SGL
	Model and Textures adjusted, removed Vanilla HAMR sight

Quad Cannon
	Massive ROF Increase
	Fixed the ability to ADS
	Added 101st Holos to the sights

Helmets
	Fixed Firestarter's Helmet not being a medical helmet
	Fixed the default pilot helmet not having thermals
	Added Shawgrass Helmet

Resupplies
	Attempted to fix HE GL rounds not showing up in resupplies

101st Holosights
	Added 101st Custom Holosights courtesy of RD501 and a lot of hard work. There is a CQC variant, a 2x-4x with CQC sights on top, and a 2x-6x with CQC sights on top. Please give feedback on the sights to CX-P Frisk

	Additionally, if you think you can do it better, also speak to CX-P Frisk, he's accepting submissions for holo sight designs.

Changelog 5.12.21
	Weapons -
		General - Most weapons we had had severe overpen issues. In order to walk back significant power creep and fix the overpen issues, our weapons have been put closer to vanilla values. Some weapons such as the DC15A recieve a substatial buff from this, where as some weapons are getting toned down to a level closer to vanilla counterparts.

		All Weapons - All weapons that did not have a flashlight before now have integrated flashlights. Press 'L' to activate by default.

		DC Series - Removal of high/medium/lows, the DC series ammo is above, and all DC series weapons can use the same basic magazine. This should allow resupply in field to be a more universal process.
			DC15A - Increased Accuracy to fit better as a battle rifle style weapon - takes standard 30 round magazines
			DC15S - Slightly decreased accuracy to better fit its idendity as a carbine, has a higher ROF in close quarters to compensate - takes 30 round standard magazines
			LE - Has access to 15 round LE Mags, and 30 round Standard Magazines
			Mod 0 - Accuracy slightly decreased to fit better as a LMG - Has access to 120 round box magazines, and 30 round Standard Magazines
			DC15S Secondary - Decreased accuracy as you are one handing a carbine, helps the weapon better fit the riot shield/CQC focused ideal.

		Boma - Magazines have been standardized to 8 round slugs and 8 round scatter, the scatter ammo no longer drops buildings as easily. Magazine weight has been adjusted.

		Gundark - Low capacity but high power sniper rifle, has 2 types of magazines, one is an overcharged magazine which has better long range capabilities.

		IR-01 Kath Hound - Has a higher ROF but weaker individual shots than the DC series, magazine capacity increased to 60 to offset this change.

		Maalras SMG - Low penetration but high ROF. Accuracy decreased to better fit the intended role as an SMG.

		SM-01 Enforcer - Low penetration but incredibly high ROF. Accuracy and damage adjusted to better fit the intended role as a PDW.

		MG5 Zakkeg - Slightly adjusted accuracy slightly to keep this as being used as a marksman rifle. It retains its 75 round magazines, however the magazine weight has been adjusted.

		Verpine DMR - Moderate Damage, high penetration rounds. The magazines only being 15 rounds have been adjusted in weight to be lighter.

		W4-LRUS - High Damage, High Velocity rounds. Intended as an allrounder 'sniper' rifle.

		XLE-47 Valken- Moderate Damage, high penetration rounds. The mag weight has been adjusted to fit its capcity.

		Zillo AT-6 - Ammunition scaled back to fit with current AT rockets.


		Pistols - These have generally been reworked to have good close range capabilities, but are no as effective at range.

		A180 - Fixed various bugs, is still in development.

		Cannok - Now has 5 Round Cylinder, can load Slugs or Scatter rounds

		DC17 - 20 Round magazine, reasonably effective at close range. Should perform relatively similar to previous interations.

		PDW Defender - Increased magazine size to 25. Weaker than the DC17 individually, but has full auto/burst to compensate.

		Westar 35 - Hits indvidually harder than DC17, but only has 7 rnd mag to compensate.

	CIS Assets
		Infantry
			B1 - They should be 1 shot headshots by everything, and soak up several rounds before going down to body shots.
			B2 - Will take several magazines to take down from DC15A, no longer unkillable. Weakpoints are the pelvis and the face.
			BX - Significantly tougher armor than b1s, but same health.
			IG88 - Invulnerable to small arms. Needs to be taken down with either concentrated fire from ADV marksman weapons, Grenade/AT launchers, or vehicles. However they are no longer unkillable death machines.
			Magnaguards - Tougher than B1's, but not by much.
		Vehicles
			Techinical - Truck with a powerful machinegun on back. Disabled by killing crew or with anything explosive.

Addons/_ammo/NewMags.hpp
	class CA_Magazine;
	class OPTRE_4Rnd_145x114_HVAP_Mag;

	class DBA_Mag_Base:CA_Magazine
	{
		author="Trip";
		scope=1;
		displayname="all round base";
		initspeed=900;
		nameSound="magazine";
		maxLeadSpeed=25;
		mass=8;
		reloadaction="";
		useactiontitle="";
		modelspecial="";
		tracersevery=1;
		picture="";
		type=256;
	};
	class  DBA_556_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="5.56mw Magazine(30rnd Blue)";
		ammo="DBA_556_blue";
		picture = "\MRC\JLTS\weapons\DC15S\data\ui\DC15S_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15S\DC15S_mag.p3d";
		mass=8;
		count=30;
		initspeed=909;
	};
	class  DBA_556_Mag_x60_Green: DBA_Mag_Base
	{
		scope=2;
		displayname="5.56mw Magazine(60rnd Green)";
		ammo="DBA_556_green";
		picture = "\MRC\JLTS\weapons\DC15S\data\ui\DC15S_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15S\DC15S_mag.p3d";
		mass=8;
		count=60;
		initspeed=909;
	};
	class  DBA_556_Mag_x300 : DBA_Mag_Base
	{
		scope = 2;
		displayname = "5.56mw Magazine(300rnd Blue)";
		ammo = "DBA_556_blue";
		picture = "\MRC\JLTS\weapons\DC15S\data\ui\DC15S_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15S\DC15S_mag.p3d";
		mass = 30;
		count = 300;
		initspeed = 909;
	};
	class DBA_65_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="6.5mw Magazine(30rnd Blue)";
		ammo="DBA_65_blue";
		picture = "\MRC\JLTS\weapons\DC15A\data\ui\DC15A_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15A\DC15A_mag.p3d";
		mass=10;
		count=30;
		initspeed=909;
	};
	class DBA_65_Mag_x45_Blue: DBA_65_Mag_Base
	{
		scope=2;
		displayname="6.5mw Magazine(45rnd Blue)";
		ammo="DBA_65_blue";
		picture = "\MRC\JLTS\weapons\DC15A\data\ui\DC15A_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15A\DC15A_mag.p3d";
		mass=15;
		count=45;
		initspeed=909;
	};
	class DBA_65_Mag_x30_Red: DBA_Mag_Base
	{
		scope=2;
		displayname="6.5mw Magazine(30rnd Red)";
		ammo="DBA_65_red";
		picture = "\MRC\JLTS\weapons\DC15A\data\ui\DC15A_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15A\DC15A_mag.p3d";
		mass=10;
		count=30;
		initspeed=909;
	};
	class DBA_65_Mag_x30_yellow: DBA_Mag_Base
	{
		scope=2;
		displayname="6.5mw Magazine(30rnd Yellow)";
		ammo="DBA_65_yellow";
		picture = "\MRC\JLTS\weapons\DC15A\data\ui\DC15A_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15A\DC15A_mag.p3d";
		mass=10;
		count=30;
		initspeed=909;
	};
	class DBA_65_Mag_x120_yellow: DBA_65_Mag_x30_yellow
	{
		scope = 2;
		mass=40; 
		displayname = "6.5mw Magazine(120rnd Yellow)";
		count = 120;
	};
	class DBA_65_Mag_x120 : DBA_65_Mag_Base
	{
		scope = 2;
		ammo = "DBA_762_base";
		picture = "\MRC\JLTS\weapons\DC15A\data\ui\DC15A_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15A\DC15A_mag.p3d";
		initspeed = 909;
		mass=40; 
		displayname = "6.5mw Magazine(120rnd Blue)";
		count = 120;
	};
	class DBA_762_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="7.62mw Magazine(30rnd Blue)";
		ammo="DBA_762_blue";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass=12;
		count=30;
		initspeed=909;
	};
	class DBA_Mag_ORBITAL : DBA_Mag_Base
	{
		scope = 2;
		displayname = "Orbital Laser Power Cell";
		ammo = "DBA_AMMO_ORBITAL";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 12;
		count = 1;
		initspeed = 909;
	};
	class DBA_65_Mag_x20 : DBA_762_Mag_Base
	{
		scope = 2;
		mass = 20;
		displayname = "7.62mw Magazine(20rnd Blue)";
		count = 20;
	};
	class DBA_762_x75_Green : DBA_Mag_Base
	{
		scope = 2;
		displayname = "7.62mw Magazine(70rnd Green)";
		ammo = "DBA_762_green";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 35;
		count = 70;
		initspeed = 909;
	};
	class DBA_762_Mag_x50 : DBA_Mag_Base
	{
		scope = 2;
		displayname = "Valken Magazine (50rnd Blue)";
		ammo = "DBA_Vulken";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 55;
		count = 50;
		initspeed = 909;
	};
	class DBA_762_Mag_x15 : DBA_Mag_Base
	{
		scope = 2;
		displayname = "7.62mw Magazine (15rnd Blue)";
		ammo = "DBA_762_blue";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 6;
		count = 15;
		initspeed = 909;
	};
	class DBA_338_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="8.6mw Magazine(15rnd Blue)";
		ammo="DBA_338_blue";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass=15;
		count=15;
		initspeed=880;
	};
	class DBA_408_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="10.4mw Magazine(10rnd Blue)";
		ammo="DBA_408_blue";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass=14;
		count=10;
		initspeed=880;
	};
	class DBA_127x99_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="12.7mw Magazine(10rnd Blue)";
		ammo="DBA_127x99_blue";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass=20;
		count=10;
		initspeed=910;
	};
	class DBA_127x99Overcharge_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="12.7mw Overcharged Magazine(5rnd Blue)";
		ammo="DBA_127x99_SLAP_blue";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass=20;
		count=5;
		initspeed=1400;
	};

	class DBA_9mm_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="9.0pw Magazine(20rnd Blue)";
		ammo="DBA_9x21_blue";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass=5;
		count=20;
		initspeed=430;
	};
	class DBA_9mm_Mag_x25 : DBA_Mag_Base
	{
		scope = 2;
		displayname = "9.0pw Magazine(25rnd Blue)";
		ammo = "DBA_9x21_blue";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass = 6;
		count = 25;
		initspeed = 430;
	};
	class DBA_9mm_Mag_x100 : DBA_Mag_Base
	{
		scope = 2;
		displayname = "9.0pw Magazine(100rnd Green)";
		ammo = "DBA_9x21_green";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass = 25;
		count = 100;
		initspeed = 430;
	};
	class DBA_45acp_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="11.43pw Magazine(20rnd Blue)";
		ammo="DBA_45acp_blue";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass=8;
		count=20;
		initspeed=254;
	};
	class DBA_45acp_Mag_x45_Green : DBA_Mag_Base
	{
		scope = 2;
		displayname = "11.43pw Magazine(45rnd Green)";
		ammo = "DBA_45acp_green";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass = 18;
		count = 45;
		initspeed = 254;
	};
	class DBA_127x33_Mag_Base: DBA_Mag_Base
	{
		scope=2;
		displayname="12.7pw Magazine(20rnd Blue)";
		ammo="DBA_127x33_blue";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass=12;
		count=20;
		initspeed=350;
	}
	class DBA_127x33_x7_Yellow : DBA_Mag_Base //Hammerhead
	{
		scope = 2;
		displayname = "12.7pw Magazine(7rnd Yellow)";
		ammo = "DBA_127x33_yellow";
		model = "\MRC\JLTS\weapons\DC17SA\DC17SA_mag.p3d";
		picture = "\MRC\JLTS\weapons\DC17SA\data\ui\DC17SA_mag_ui_ca.paa";
		mass = 12;
		count = 7;
		initspeed = 350;
	};

	class DBA_12gaScatter_Mag_Base: DBA_Mag_Base //Boma
	{
		scope=2;
		displayname="12sw Scatter Magazine(8rnd Green)";
		ammo="DBA_12gaPellets_Base";
		picture = "\MRC\JLTS\weapons\EPL2\data\ui\EPL2_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\EPL2\EPL2_mag.p3d";
		mass=8;
		count=8;
		initspeed=400;
	};

	class DBA_12gaScatter_x5_Blue : DBA_Mag_Base //Cannok
	{
		scope = 2;
		displayname = "12sw Scatter Magazine(5rnd Green)";
		ammo = "DBA_12gaPellets_Base";
		picture = "\MRC\JLTS\weapons\EPL2\data\ui\EPL2_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\EPL2\EPL2_mag.p3d";
		mass = 5;
		count = 5;
		initspeed = 400;
	};

	class DBA_12gaSlug_Mag_Base: DBA_Mag_Base //Boma
	{
		scope=2;
		displayname="12sw Slug Magazine(8rnd Green)";
		ammo="DBA_12gaSlug_green";
		picture = "\MRC\JLTS\weapons\EPL2\data\ui\EPL2_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\EPL2\EPL2_mag.p3d";
		mass=8;
		count=8;
		initspeed=400;
	};
	class DBA_12gaSlug_x5_Blue: DBA_Mag_Base //Cannok
	{
		scope=2;
		displayname="12sw Slug Magazine(5rnd Green)";
		ammo="DBA_12gaSlug_green";
		picture = "\MRC\JLTS\weapons\EPL2\data\ui\EPL2_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\EPL2\EPL2_mag.p3d";
		mass=5;
		count=5;
		initspeed=400;
	};

	class DBA_20mm_x7_Base : OPTRE_4Rnd_145x114_HVAP_Mag
	{
		scope = 2;
		displayname = "20mw Magazine(7rnd Orange)";
		picture = "\MRC\JLTS\weapons\EPL2\data\ui\EPL2_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\EPL2\EPL2_mag.p3d";
		mass = 56;
		count = 7;
		initspeed = 1400;
		ammo = "DBA_AntiTank_base";
	};
	class DBA_EMP_x1_Base : OPTRE_4Rnd_145x114_HVAP_Mag
	{
		scope = 2;
		displayname = "Antivehicle EMP (1rnd Orange)";
		picture = "\MRC\JLTS\weapons\EPL2\data\ui\EPL2_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\EPL2\EPL2_mag.p3d";
		mass = 44;
		count = 1;
		initspeed = 1400;
		ammo = "DBA_EMP_base";
	};

	
	// Imperial stuff


	class DBA_E11D_Magazine : DBA_Mag_Base
	{
		scope = 2;
		displayname = "E11D Magazine (25rnd Blue)";
		ammo = "E11D_Base_Ammo";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 55;
		count = 25;
		initspeed = 909;
	};

	//Rebel stuff
	class DBA_P7_Magazine : DBA_Mag_Base
	{
		scope = 2;
		displayname = "P7 Magazine (7rnd Blue)";
		ammo = "P7_Base_Ammo";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 55;
		count = 7;
		initspeed = 909;
	};
	// memes

	class DBA_LegoGun_Magazine : DBA_Mag_Base
	{
		scope = 2;
		displayname = "Lego Gun Magazine (120rnd Blue)";
		ammo = "DBA_LegoGun_ammo";
		picture = "\MRC\JLTS\weapons\DC15x\data\ui\DC15X_mag_ui_ca.paa";
		model = "\MRC\JLTS\weapons\DC15x\DC15x_mag.p3d";
		mass = 10;
		count = 120;
		initspeed = 909;
	};